File: LeetCode_3217/Main.py
# Definition for singly-linked list.
class ListNode(object):
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution(object):
    def modifiedList(self, nums, head):
        """
        :type nums: List[int]
        :type head: Optional[ListNode]
        :rtype: Optional[ListNode]
        """

        cnt = dict()
        for num in nums:
            if num in cnt:
                cnt[num] += 1
            else:
                cnt[num] = 1
        
        result = resultHead = None

        while head:
            if not head.val in cnt:
                if resultHead:
                    result.next = head
                    result = result.next
                else:
                    result = resultHead = head
            head = head.next
        
        result.next = None
        return resultHead

File: LeetCode_3217/Main.cpp
#include<bits/stdc++.h>
using namespace std;



// Definition for singly-linked list.

struct ListNode {
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};

class Solution {
public:
    ListNode* modifiedList(vector<int>& nums, ListNode* head) {
        ListNode* result = nullptr;
        ListNode* resultHead = nullptr;

        map<int, int> cnt;
        for (int x : nums) {
            cnt[x] += 1;
        }

        while (head) {
            if (!cnt[head->val]) {
                if (resultHead == nullptr) {
                    resultHead = head;
                    result = head;
                } else {
                    result->next = head;
                    result = result->next;
                }
            }
            head = head->next;
        }
        result->next = nullptr;
        return resultHead;
    }
};